### ISSUE TITLE
Encountering issues when creating a VPC and Firewall rules using the Terraform GCP blueprint.

### ISSUE THREAD

**Issue by rahul-tiple** [terraform-google-modules/terraform-google-network#542]
### TL;DR

When creating VPC and Firewall rules using "Terraform blueprints and modules for Google Cloud," the VPC is successfully created. However, when attempting to create the Firewall at the same time, an error occurs: "Network name already created." This error found because I am using the same source, "terraform-google-modules/network/google." Additionally, when using the source "terraform-google-modules/network/google//modules/firewall-rules," a version conflict error occurs.

How to set the default variables in the firewall_rules variable.tf file:

### Expected behavior

https://github.com/terraform-google-modules/terraform-google-network/blob/master/main.tf

### Observed behavior

Error: Invalid type specification
  on .terraform/modules/firewall_rules/modules/firewall-rules/variables.tf line 32, in variable "rules":
  32:     description             = optional(string, null)
Keyword "optional" is not a valid type constructor.

and

version conflict

### Terraform Configuration

```hcl
variable "rules" {
  description = "This is DEPRICATED and available for backward compatiblity. Use ingress_rules and egress_rules variables. List of custom rule definitions"
  type = list(object({
    name                    = string
    description             = optional(string, null)
    direction               = optional(string, "INGRESS")
    disabled                = optional(bool, null)
    priority                = optional(number, null)
    ranges                  = optional(list(string), [])
    source_tags             = optional(list(string))
    source_service_accounts = optional(list(string))
    target_tags             = optional(list(string))
    target_service_accounts = optional(list(string))

    allow = optional(list(object({
      protocol = string
      ports    = optional(list(string))
    })), [])
    deny = optional(list(object({
      protocol = string
      ports    = optional(list(string))
    })), [])
    log_config = optional(object({
      metadata = string
    }))
  }))
  default = [
    {
      name                    = "allow-ssh-ingress"
      description             = null
      direction               = "INGRESS"
      priority                = null
      destination_ranges      = ["10.0.0.0/8"]
      source_ranges           = ["0.0.0.0/0"]
      source_tags             = null
      source_service_accounts = null
      target_tags             = null
      target_service_accounts = null
      allow = [{
        protocol = "tcp"
        ports    = ["22"]
      }]
      deny = []
      log_config = {
          metadata = "INCLUDE_ALL_METADATA"
      }
    }
  ]
}
```

### Terraform Version

```sh
Your version of Terraform is out of date! The latest version
is 1.7.0. You can update by downloading from https://www.terraform.io/downloads.html
Terraform v0.13.0
```

### Additional information

Please explain how to use the Terraform GCP blueprint and set the default variables in the firewall_rules variable.tf file.

**Comment by imrannayer** [comment]
@rahul-tiple it seems like you are using older version of Terraform (0.13). Module support TF version 1.3+

https://github.com/terraform-google-modules/terraform-google-network?tab=readme-ov-file#compatibility
